Lib/test/test_winapi.py

```diff
@@ -127,3 +127,37 @@ def test_getshortpathname(self):
 
         # Should contain "PROGRA~" but we can't predict the number
         self.assertIsNotNone(re.match(r".\:\\PROGRA~\d", actual.upper()), actual)
+
+    def test_namedpipe(self):
+        # Deliberately complex name to flush out encoding errors
+        pipe_name = os.urandom(64).replace(b'\0', b'.').decode('latin-1')
+        pipe_name = rf"\\.\pipe\LOCAL\{pipe_name}"
+
+        # Pipe does not exist, so this raises
+        with self.assertRaises(FileNotFoundError):
+            _winapi.WaitNamedPipe(pipe_name, 0)
+
+        pipe = _winapi.CreateNamedPipe(
+            pipe_name,
+            _winapi.PIPE_ACCESS_DUPLEX,
+            8, # 8=PIPE_REJECT_REMOTE_CLIENTS
+            2, # two instances available
+            32, 32, 0, 0)
+        self.addCleanup(_winapi.CloseHandle, pipe)
+
+        # Pipe instance is available, so this passes
+        _winapi.WaitNamedPipe(pipe_name, 0)
+
+        with open(pipe_name, 'w+b') as pipe2:
```

Comment:
I guess checking `open()` suffices. The old broken implementation passed a UTF-8 encoded string to `CreateNamedPipeA()`, which decoded it using the process ANSI code page. If a legacy ANSI code page was in use, the result was mojibake for non-ASCII characters. That was consistent for `_winapi.WaitNamedPipe()`, and it would still be if they both regressed to using ANSI. OTOH, `open()` uses the wide-character name.